code/modules/mob/living/living_defense.dm

```diff
@@ -284,6 +284,55 @@
 
 	return G
 
+/mob/living/proc/surgery_attack_chain(mob/living/user, obj/item/I)
+	if(user == src || user.a_intent != INTENT_HELP)
+		return FALSE
+	if(can_be_operated_on())
+		var/datum/surgery/S = get_or_initiate_surgery(user)
+		if(S)
+			return surgery_act(user, I, S)
+	return FALSE
+
+
+//check if mob is lying down on something we can operate him on.
+/mob/living/proc/can_be_operated_on()
+	if(locate(/obj/machinery/optable, loc) && (lying || resting))
+		return TRUE
+	if(locate(/obj/structure/bed, loc) && (buckled || lying || IsWeakened() || stunned || paralysis || sleeping || stat))
+		return TRUE
+	if(locate(/obj/structure/table, loc) && (lying || IsWeakened() || stunned || paralysis || sleeping || stat))
```

Comment:
I think you could use `incapacitated()` here to cut down on copy paste, since it covers *most* of the checks you want.

Reply:
will see about changing that mess there. Looks like it can be simplified by a lot

Reply:
Changed it to just check if you're lying. I don't see any reason to have the other checks since all cases I can think of will make the `lying` variable have a value.